[NPQ-3456] Added C4 diagram (#3483)
DFE-Digital/npq-registration: docs/c4container.md

[< Back to Navigation](../README.md)

# C4 Container model

> Internally we have Web and Worker Kubernetes containers nodes, with the web

Reviewer: ```suggestion
Internally we have web and worker Kubernetes containers nodes, with the web
```
web is not really capitalised any more, and worker definitely shouldn't be

> containers accessed via a CDN backed onto an NGINX kubernetes ingress. These are

Reviewer: ```suggestion
containers accessed via a CDN backed onto an nginx kubernetes ingress. These are
```
nginx is officially lowercase

> backed by Azure managed Postgres, Redis and Blob storage services.

Reviewer: ```suggestion
backed by Azure managed Postgres, Redis and blob storage services.
```
blob

Externally we integrate with

> * the TeacherAuth service provided by the TRS service, which itself then integrates with the GovUK OneLogin service

Reviewer: ```suggestion
* the TeacherAuth service provided by the TRS service, which itself then integrates with the GOV.UK One Login service
```
officially GOV.UK One Login

* we provide certificate information to the TRS service via an internal API
* we request TRN allocation for new TeacherAuth users via a TRS API, via a per-user OAuth access token

> * we are notified of user changes via signed Webhooks we provide to TRS

Reviewer: ```suggestion
* we are notified of user changes via signed webhooks we provide to TRS
```
webhooks

> * we send emails out using the GovUK Notify service

Reviewer: ```suggestion
* we send emails out using the GOV.UK Notify service
```

Below is a C4 Container diagram for the NPQ service

```mermaid
C4Container
  title Cloud Infrastructure Platform

  System_Ext("lp1", "Lead Provider 1", "Lead Provider API", "Bearer Token secured API for Lead Providers to interact with NPQ service")
  System_Ext("lp2", "Lead Provider 2", "Lead Provider API", "Bearer Token secured API for Lead Providers to interact with NPQ service")

>   Person(citizen, "User registering for NPQ", "Citizen")

Reviewer: Is 'Citizen' the right word here? I guess everyone is a citizen of somewhere, but this implies a UK citizen, which is not always the case.
Maybe 'Participant'?

Author: No strong feelings but I believe its the term used with C4 diagrams to make clear its a member of the public rather than an 'internal user' - could change it to participant though :thinking:
